### PR TITLE
Let renovatebot update Go toolchain version in a single PR

### LINKED ISSUE #24806: Let renovatebot update Go toolchain in a single PR
In the past, we updated the Go toolchain version used for building Cilium manually by setting the new version in `GO_VERSION` and then running `make update-go-version`. Since we introduced renovatebot, the golang base docker images previously updated by that make target are now updated by the bot, see https://github.com/cilium/cilium/pull/22115. However, some places (e.g. `.travis.yaml` https://github.com/cilium/cilium/blob/22161112e06f215a5af9485c05489eba5aa21504/.travis.yml#L6) still need to be updated manually using the make target (see https://github.com/cilium/cilium/pull/22589 for an example of such an update). Also the automated updates created by the bot are spread across multiple PRs.

This issue is about migrating the remaining updates from the `update-go-version` target to renovatebot and changing renovatebot to group all Go version updates into a single PR akin to the cilium-cli and hubble updates (ref. https://github.com/cilium/cilium/pull/24725 and https://github.com/cilium/cilium/pull/24637).

### PR DESCRIPTION
**Pull request by tklauser**
Migrate the remaining Go toolchain updates from the `update-go-version` target to renovatebot and change renovatebot to group all Go version updates into a single PR.

See individual commits and #24806 for details.

Example PR: https://github.com/tklauser/cilium/pull/504

Fixes: #24806

### CODE REVIEW

**Review by bimmlerd — APPROVED** [review]
very nice. Minor nit, the first commit msg you state
> [...] GO_VERSION (currently Go 1.12.1), thus the second condition (version >= go 1.13)

where I think you meant to say "currently 1.20.1".

File: .github/renovate.json5 [hunk 1]
```diff
@@ -229,6 +232,13 @@
         "quay.io/cilium/hubble"
       ],
     },
+    {
+      "groupName": "Go",
+      "matchDepNames": [
+        "go",
+        "docker.io/library/golang"
+      ]
+    },
```
Comment (michi-covalent) [inline .github/renovate.json5]: could we do:
```suggestion
    {
      "groupName": "Go",
      "matchDepNames": [
        "go",
        "docker.io/library/golang"
      ],
      "schedule": [
        "on friday"
      ]
    },
```
image SHAs for `docker.io/library/golang` keeps changing even for the same patch version (see https://github.com/cilium/cilium/pull/24846), and renovate keeps opening new pull requests 🥺

**Review by chancez — APPROVED** [review]
This is awesome.

**Review by aanm — APPROVED** [review]
:tada:

**Review by sayboras — APPROVED** [review]

**Review by zacharysarah — COMMENTED** [review]
LGTM from a docs perspective.

File: .github/renovate.json5 (hunk 1 above)
Comment (tklauser) [reply .github/renovate.json5]: Good point, done.

**Comment by tklauser** [comment]
Rebased to resolve merge conflicts and to address David's and Michi's review comments.

**Review by michi-covalent — APPROVED** [review]

**Pull request merged**